zb-db/src/main/java/io/zeebe/db/impl/rocksdb/transaction/ZeebeTransactionDb.java

```diff
@@ -104,7 +103,9 @@ private static long getNativeHandle(final RocksObject object) {
   private final DirectBuffer keyViewBuffer = new UnsafeBuffer(0, 0);
   private final DirectBuffer valueViewBuffer = new UnsafeBuffer(0, 0);
 
-  private final ExpandableArrayBuffer prefixKeyBuffer = new ExpandableArrayBuffer();
+  private int activePrefixIterations = 0;
+  private final ExpandableArrayBuffer[] prefixKeyBuffers =
```

Comment:
You could also use a Queue from which you pop/push, so that you don't need to keep track of a counter, just throw an exception if the queue is empty (so not a blocking queue).

Reply:
But I would say that's not the main problem so arrays + counter is probably fine for now